src/Neo/SmartContract/Native/Notary.cs

```diff
@@ -223,8 +221,12 @@ private async ContractTask<bool> Withdraw(ApplicationEngine engine, UInt160 from
             if (Ledger.CurrentIndex(engine.Snapshot) < deposit.Till) throw new InvalidOperationException(string.Format("Can't withdraw before {0}", deposit.Till));
             RemoveDepositFor(engine.Snapshot, from);
 
-            await engine.CallFromNativeContract(Hash, GAS.Hash, "transfer", Hash.ToArray(), receive.ToArray(), deposit.Amount, null);
-            return true;
+            // TODO: this code is invalid and doesn't work as expected. We need to throw an exception if 'transfer' call returns 'false'.
+            return engine.CallFromNativeContract<bool>(Hash, GAS.Hash, "transfer", Hash.ToArray(), receive.ToArray(), deposit.Amount, null);
+            // if (!transferOK) {
+            //     throw new InvalidOperationException(string.Format("Transfer to {0} has failed", receive.ToString()));
+            // }
+            //return true;
```

Comment:
Could anyone help me with this part of code, please? The desired behaviour is to check the return value of this `transfer` call, and if it's `false`, then we need to throw an exception. But I'm not sure how to properly retrieve the boolean result of `engine.CallFromNativeContract<bool>` call.

The result may be tested via `Check_ExpirationOf` unit-test.

Reply:
@AnnaShaleva please check it

Reply:
It should work with ContractTask<T>... :S

Reply:
Good, thank you! Will check and add some more tests in nearest future.

Reply:
I found the bug

Reply:
I found the bug, i will open a pr today

Reply:
https://github.com/neo-project/neo/pull/3190